github/resource_github_repository_environment_deployment_policy.go

```diff
@@ -0,0 +1,157 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/v50/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+func resourceGithubRepositoryEnvironmentDeploymentPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubRepositoryEnvironmentDeploymentPolicyCreate,
+		Read:   resourceGithubRepositoryEnvironmentDeploymentPolicyRead,
+		Update: resourceGithubRepositoryEnvironmentDeploymentPolicyUpdate,
+		Delete: resourceGithubRepositoryEnvironmentDeploymentPolicyDelete,
+		Schema: map[string]*schema.Schema{
```

Comment:
Preamble: Not an expert in implementing a terraform provider.

The GitHub api that exposes a get deployment branch policy requires an ID, this ID is not exposed anywhere in the UI so the user has no way (unless it's queries the GitHub api) to know such ID.

One option I was thinking about was to read all the deployment branch policies and then filter by branch pattern but I thought such an implementation to be overkill, especially because this resource only contains a branch name.

What are your thoughts?

Reply:
I think it's still worth implementing with an ID even if it's not clearly visible in the GitHub UI; there's still value in it.

Reply:
I just implemented support for import